control/trajectory_follower/obstacle_collision_checker/src/obstacle_collision_checker_node/obstacle_collision_checker_node.cpp

```diff
@@ -0,0 +1,335 @@
+    RCLCPP_INFO_THROTTLE(
+      this->get_logger(), *this->get_clock(), 5000 /* ms */,
+      "waiting for obstacle_pointcloud msg...");
+    return false;
+  }
+
+  if (!obstacle_transform_) {
+    RCLCPP_INFO_THROTTLE(
+      this->get_logger(), *this->get_clock(), 5000 /* ms */, "waiting for obstacle_transform...");
+    return false;
+  }
+
+  if (!reference_trajectory_) {
+    RCLCPP_INFO_THROTTLE(
+      this->get_logger(), *this->get_clock(), 5000 /* ms */,
+      "waiting for reference_trajectory msg...");
+    return false;
+  }
+
+  if (!predicted_trajectory_) {
+    RCLCPP_INFO_THROTTLE(
+      this->get_logger(), *this->get_clock(), 5000 /* ms */,
+      "waiting for predicted_trajectory msg...");
+    return false;
+  }
+
+  if (!current_twist_) {
+    RCLCPP_INFO_THROTTLE(
+      this->get_logger(), *this->get_clock(), 5000 /* ms */, "waiting for current_twist msg...");
+    return false;
+  }
+
+  return true;
+}
+
+bool ObstacleCollisionCheckerNode::isDataTimeout()
+{
+  const auto now = this->now();
+
+  constexpr double th_pose_timeout = 1.0;
+  const auto pose_time_diff = rclcpp::Time(current_pose_->header.stamp).seconds() - now.seconds();
+  if (pose_time_diff > th_pose_timeout) {
+    RCLCPP_WARN_THROTTLE(
+      this->get_logger(), *this->get_clock(), 1000 /* ms */, "pose is timeout...");
+    return true;
+  }
+
+  return false;
+}
+
+void ObstacleCollisionCheckerNode::onTimer()
+{
+  current_pose_ = self_pose_listener_->getCurrentPose();
+  if (obstacle_pointcloud_) {
+    const auto & header = obstacle_pointcloud_->header;
+    obstacle_transform_ = transform_listener_->getTransform(
+      "map", header.frame_id, header.stamp, rclcpp::Duration(0.01));
+  }
+
+  if (!isDataReady()) {
+    return;
+  }
+
+  if (isDataTimeout()) {
+    return;
+  }
+
+  input_.current_pose = current_pose_;
+  input_.obstacle_pointcloud = obstacle_pointcloud_;
+  input_.obstacle_transform = obstacle_transform_;
+  input_.reference_trajectory = reference_trajectory_;
+  input_.predicted_trajectory = predicted_trajectory_;
+  input_.current_twist = current_twist_;
+
+  output_ = obstacle_collision_checker_->update(input_);
+
+  updater_.force_update();
+
+  debug_publisher_->publish("marker_array", createMarkerArray());
+
+  time_publisher_->publish(output_.processing_time_map);
+}
+
+rcl_interfaces::msg::SetParametersResult ObstacleCollisionCheckerNode::paramCallback(
+  const std::vector<rclcpp::Parameter> & parameters)
+{
+  rcl_interfaces::msg::SetParametersResult result;
+  result.successful = true;
+  result.reason = "success";
+
+  Param param;
+  try {
+    update_param(parameters, "delay_time", param.delay_time);
+    update_param(parameters, "footprint_margin", param.footprint_margin);
+    update_param(parameters, "max_deceleration", param.max_deceleration);
+    update_param(parameters, "resample_interval", param.resample_interval);
+    update_param(parameters, "search_radius", param.search_radius);
+    param_ = param;
```

Comment:
Do you need to set parameter by `obstacle_collision_checker_->setParam(param_);`?

Reply:
that's true !